boards: seeed: xiao_nrf54l15: update Kconfig and DTS for ROM start of…

Bohrium332 opened the pull request:
Set the starting address to 0 if the partition manager is enabled.
Designate the flash controller as rram_controller. 
Enable IEEE 802.15.4 wireless communication support.

nordicjm commented on boards/seeed/xiao_nrf54l15/Kconfig.defconfig:
```diff
@@ -5,4 +5,8 @@ if BOARD_XIAO_NRF54L15_NRF54L15_CPUAPP
 	default BT
 
+config ROM_START_OFFSET
+	default 0 if PARTITION_MANAGER_ENABLED
```
this is not acceptable

nordicjm commented on boards/seeed/xiao_nrf54l15/xiao_nrf54l15_nrf54l15_cpuapp.dts:
```diff
@@ -44,9 +44,11 @@
 		zephyr,code-partition = &slot0_partition;
 		zephyr,sram = &cpuapp_sram;
 		zephyr,flash = &cpuapp_rram;
+		zephyr,flash-controller = &rram_controller;
```
remove `Kconfig and` from commit title, there is no Kconfig update here

Bohrium332 committed:
boards: seeed: xiao_nrf54l15: update DTS

Designate the flash controller as rram_controller.
Enable IEEE 802.15.4 wireless communication support.

Signed-off-by: Zhida Zhang <2759897880@qq.com>

nordicjm commented:
> This PR is actually missing the ROM start for partition manager

Correct, partition manager has nothing to do with zephyr
